### ISSUE TITLE
Feature Request : Gradients for tf.contrib.image.rotate()

### ISSUE THREAD

**Issue by AshishBora** [tensorflow/tensorflow#9423]
I wish to propagate gradients through image rotation. This is currently not supported:

```
import tensorflow as tf
import numpy as np

images = tf.zeros(shape=[5, 10, 10, 1])
angles = 2 * np.pi * np.random.random(5) - np.pi
out = tf.contrib.image.rotate(images, angles)
out_sum = tf.reduce_sum(out)
images_grad = tf.gradients(out_sum, [images])

print images_grad
```

gives

```[None]```

**Comment by AshishBora** [comment]
I think the gradient with respect to `images` can be easily implemented by applying reversed rotation on the gradient coming from top. i.e. if `out_grad` is gradient with respect to `out`, then
```
images_grad = tf.contrib.image.rotate(out_grad, -angles)
```

Gradient with respect to angles might be trickier. However, I do not need this functionality.

**Comment by aselle** [comment]
If you are not up to making a full patch (But please do if you can).. you can just define a gradient for the op ad-hoc in your code using the gradient_override_map... i.e.

```python
@tf.RegisterGradient(‘CustomSquare’)
def custom_square_grad(op, grad):
  # …

c = tf.constant(5.0)
s1 = tf.square(c)  # Uses default gradient for square
with graph.gradient_override_map({‘Square’: ‘CustomSquare’}):
  s2 = tf.square(c)  # Uses _compute_square_grad
```

**Comment by sebastian-schlecht** [comment]
I gave this one a shot - as a first time contributor I guess some feedback would be really appreciated. Can put up a PR soon.

**Comment by alxy** [comment]
I am interested in the gradients with respect to the angle(s) as well. Is this theoretically possible? As I am just getting started with TF, I'm not really sure how this could possibly be implemented...

**Comment by zcz004** [comment]
Can tf.contrib.image.rotate() be backpropagated at the moment? I have tried training with this op involved in my network architecture but failed because of it Is it possible to add a backprop feature to it? 
(Windows 7, gpu)
